### DISCUSSION TITLE
HTML export: Linked documents open in the same page

### DISCUSSION THREAD

**Discussion by aloboa** [zim-desktop-wiki/zim-desktop-wiki#2560] [Q&A]
I have images in Zim with a link to a video or HTML page.
When I export to HTML, the linked video or pages open in the same tab, which
is confusing for readers as they get lost.
Is there a way to enforce that, by default, linked documents open in a different window or tab?

**Comment by introt** [comment]
Short answer: No

Long answer: HTML links have the [`target` attribute](https://www.w3schools.com/html/html_links.asp) for that, but applying that to just linked documents requires either adding some JavaScript into the export template that adds the attribute to external links (eg. ones that don't lead to html files on the same host) or adding the option to Zim's exporter (shouldn't be very hard, perhaps a Good First Issue)

If you don't want to open any other pages in the same tab, you can just add the wanted `target` attribute to _every_ link by adding the following JavaScript to your template, eg.:

```html
<!-- add this to the <head> -->
<script>
function openLinksInNewTabs() {
  links = document.getElementsByTagName('a')

  for (i in links) {
    links[i].target = '_blank'
}
</script>

<!-- modify the <body> tag -->
<body onload="openLinksInNewTabs();">
```

The large language models are pretty good at this stuff, feel free to ask one to expand the example to eg. exclude links to other pages or inside the index.

**Reply by aloboa** [reply]
Thanks.
It sounds like this could very well be an option at exporting to HTML, right? 
If so, I will add it as a feature request.

**Reply by introt** [reply]
Yes, it'd make sense as a template option imo.

**Reply by bttrx** [reply]
Additional note: Using "target _blank" is considered bad style, because you **force** the reader to read the linked page in a new tab. I think, it would be better to teach readers from the beginning, how to open a link in a new separate tab or window, i.e., how to use a context menu or the third mouse button.

**Reply by introt** [reply]
To clarify, imo using target _blank makes sense for (1)external links and (2)attachments, as (1)many sites have a nasty habit of abusing the history API, and (2)downloading files on certain browsers also tends to mess up the back button navigation, or at the very least your position on the page.
